docs/tutorials/skeletonization.md

```diff
@@ -0,0 +1,229 @@
+---
+title: Skeletonization
+---
+
+The skeleton service enables you to generate and retrieve skeletons from the server. This saves you the trouble of performing skeletonization routines on your local machine, while also saving you the trouble of understanding many of the details inside of skeletonization. Furthermore, the server offers the ability to generate multiple skeletons simultaneously, in parallel. And lastly, the server caches each skeleton that is generated so that it only ever needs to be generated once. All subsequent requests for the same skeleton will simply return the cached skeleton with notable turn-around-time improvements.
+
+## Initializing the client
+
+The simplest way to initialize the CAVEclient is by merely providing the datastack of interest:
+
+```python
+client = cc.CAVEclient(<datastack_name>)
+```
+
+With a CAVEclient built, you can now investigate the current build version of the SkeletonClient:
+
+```python
+client.skeleton.get_version()
+```
+
+And you can see a list of available skeleton versions. In most cases you will want to use the highest (most recent) version provided.
+
+```python
+client.skeleton.get_versions()
+```
+
+## Retrieving a skeleton
+
+Retrieve a skeleton using `get_skeleton()`. The simplest usage is:
+
+```python
+sk = client.skeleton.get_skeleton(
+    <root_id>,
+    output_format=<output_format>,
+)
+```
+
+where the availale output_formats (described below) are:
+
+* none (default if unspecified)
+* h5
+* swc
+* swccompressed
+* json
+* jsoncompressed
+* arrays
+* arrayscompressed
+* precomputed
+
+If the skeleton doesn't exist in the server cache, it may take 20-60 seconds to generate the skeleton before it is returned. This function will block during that time. Any subsequent retrieval of the same skeleton should go very quickly however.
+
+To specify a nondefault skeleton version:
+
+```python
+sk = client.skeleton.get_skeleton(
+    <root_id>,
+    skeleton_version=<sk_version>,
+    output_format=<output_format>,
+)
+```
+
+To specify a specific datastack:
+
+```python
+sk = client.skeleton.get_skeleton(
+    <root_id>,
+    datastack_name=<datastack_name>,
+    skeleton_version=<sk_version>,
+    output_format=<output_format>,
+)
+```
+
+## Peering into the contents of the cache
```

Comment:
let's put this at the bottom... i think it's confusing an irrelevant for most users compare to the exists call.